File: benchmarks/pandas/bench_series_rank.py
"""Benchmark: Series.rank — rank 100k elements"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = np.sin(np.arange(ROWS) * 0.03) * 1000
s = pd.Series(data)

for _ in range(WARMUP):
    s.rank()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.rank()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "series_rank",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_series_rank.ts
/**
 * Benchmark: Series.rank — rank 100k elements
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Float64Array.from({ length: ROWS }, (_, i) => Math.sin(i * 0.03) * 1000);
const s = new Series(data);

for (let i = 0; i < WARMUP; i++) {
  s.rank();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.rank();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "series_rank",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
